lib/red_cross.rb

```diff
@@ -1,26 +1,36 @@
 require 'red_cross/version'
 require 'red_cross/configuration'
 require 'segment'
+require 'typhoeus'
 require 'resque/plugins/red_cross_async'
 require 'resque'
+require 'red_cross/trackers/base'
+require 'red_cross/trackers/segment_tracker'
+require 'red_cross/trackers/http_tracker'
 
 module RedCross
     class << self
-
       def track(attrs, topic = "")
-        Configuration.segment.track(attrs)
+        Configuration.tracker.track(attrs)
       end
 
       def identify(attrs, topic = "")
-        Configuration.segment.identify(attrs)
+        Configuration.tracker.identify(attrs)
       end
 
       def flush
-        Configuration.segment.flush
+        Configuration.tracker.flush
       end
 
       def monitor(attrs)
         []
       end
+
+      def method_missing(m, *args, &block)
+        match = /(.*?)_track/.match(m.to_s)
```

Comment:
@avichay we can do it like this but this way the method is not dynamic, were sending only :track